File: grupa C/olimpiada/nacionalen/2009/2009-05-05-studentsC/submitsC/Log31004/money.pas
(*
TASK:MONEY
LANG:PASCAL
*)

program moneymoneymoney;
var n,i,fo:longint;
    income:array[1..200000] of longint;

procedure order;
var aux, sch : longint;
begin
  repeat
    sch:=0;
    for i := 1 to n-1 do
      if income[i]<income[i+1] then
        begin
          sch:=1;
          aux:=income[i];
          income[i]:=income[i+1];
          income[i+1]:=aux;
        end;
  until sch=0;
end;

begin
{  assign(input,'money.in'); reset(input);
  assign(output,'money.out'); rewrite(output); }
  readln(n);
  for i := 1 to n do read(income[i]);
  order;
  fo:=1;
  while income[fo+1]+income[fo]>income[1] do
    inc(fo);
  if fo=2 then fo:=1;
  writeln(fo);
 { close(input);
  close(output);}
end.

File: grupa C/olimpiada/nacionalen/2009/2009-05-05-studentsC/submitsC/Log31018/money.cpp
/*
TASK:money
LANG:C++
*/
#include<iostream>
using namespace std;
int n,a[200001];
int i=1,l;
int f()
{
    while(a[i]<=a[n]/2)
    {
        i++;
    }
    if(a[i]+a[i-1]>a[n]) return n-i+2;
    else return n-i+1;
}

int f1(int m)
{
    while(a[i]>=a[m]/2)
    {
        i--;
    }
    if(a[i]+a[i+1]>a[m]) return m-i+1;
    else return m-i;
}

main()
{
    cin>>n;
    for(int j=1; j<=n; j++)
    {
        cin>>a[j];
    }
    sort(a+1,a+n+1);
    l=f();
    for(int d=n-1;d>=3;d--)
    {
        int z=f1(d);
        if(l<z) {
            l=z;
            }
        }
    if(l==2) cout<<"1";
    else cout<<l;
    cout<<endl;
    return 0;
    }
